### PR TITLE
Evac query performance

### LINKED ISSUE #244: Check performance on long list pages
Check queries for:

- [ ] Service Request list 
- [ ] Dispatch Assignment list
- [ ] Owner List

Add server-side pagination if necessary.

### PR DESCRIPTION
**Pull request by rmeyers4**
Closes #244

### CODE REVIEW

**Comment by rmeyers4** [comment]
@AlexMountain @trevorskaggs : Ready for review. I tried to click as many buttons as I could find and nothing I found returns a 500. I think I fixed most places where we needed prefetching and such and we  checked that the number of queries for an endpoint doesn't increase when returning a single object vs. list of objects. 

Main things to be on the lookout for are missing information (if I accidentally dropped a needed serializer field) and object updates. I painstakingly learned that after updating an object and running `get_queryset`, DRF drops the prefetched objects before serializing, so prefetched fields saved using `to_attr` won't be present for serialization. I think I fixed logic where necessary, but again, may have missed something!

File: evac/serializers.py [hunk 1]
```diff
@@ -44,17 +44,26 @@ class Meta:
...
+        #TODO
+        assigned_evac = EvacAssignment.objects.filter(service_requests=obj, end_time__isnull=True).values('id', 'start_time', 'end_time').first()
+        if assigned_evac:
+            return assigned_evac
+        return EvacAssignment.objects.filter(service_requests=obj, end_time__isnull=False).values('id', 'start_time', 'end_time').first()
 
     class Meta:
         model = ServiceRequest
         fields = ['id', 'latitude', 'longitude', 'full_address', 'followup_date', 'status',
-        'injured', 'accessible', 'turn_around', 'animals', 'reported_animals', 'sheltered_in_place', 'unable_to_locate', 'aco_required',
```
Comment (AlexMountain) [inline evac/serializers.py:56]: 'injured', 'reported_animals', 'sheltered_in_place', 'unable_to_locate', 'aco_required' are all being used...

see https://github.com/trevorskaggs/shelterly/blob/master/frontend/src/dispatch/Dispatch.js#L120 for an example

File: people/serializers.py [hunk 2]
```diff
@@ -61,24 +61,27 @@ class PersonSerializer(SimplePersonSerializer):
...
     animals = serializers.SerializerMethodField()
-    request = serializers.SerializerMethodField()
+    # request = serializers.SerializerMethodField()
     action_history = serializers.SerializerMethodField()
 
     def get_animals(self, obj):
         from animals.serializers import ModestAnimalSerializer
-        return ModestAnimalSerializer(obj.animal_set.exclude(status='CANCELED') | obj.animals.exclude(status='CANCELED'), many=True).data
+        if hasattr(obj, 'animals'):
+            return ModestAnimalSerializer(obj.animals, many=True).data
```
Comment (AlexMountain) [inline people/serializers.py:70]: This should be `reporter_animals` based on the change to animal models above?

File: evac/serializers.py [hunk 3]
```diff
@@ -44,17 +44,26 @@ class Meta:
...
+class DispatchServiceRequestSerializer(BarebonesServiceRequestSerializer):
 
-    animals = ModestAnimalSerializer(source='animal_set', many=True, read_only=True)
+    animals = SimpleAnimalSerializer(source='animal_set', many=True, read_only=True)
     owner_contacts = OwnerContactSerializer(source='ownercontact_set', many=True, required=False, read_only=True)
     owner_objects = SimplePersonSerializer(source='owners', many=True, required=False, read_only=True)
+    visit_notes = VisitNoteSerializer(source='visitnote_set', many=True, required=False, read_only=True)
+    latest_evac = serializers.SerializerMethodField()
+
+    def get_latest_evac(self, obj):
```
Comment (AlexMountain) [inline evac/serializers.py:55]: I believe I made updates such that we no longer need `latest_evac`.  OK with removing?

File: evac/models.py [hunk 4]
```diff
@@ -37,8 +37,8 @@ class Meta:
 
 class AssignedRequest(models.Model):
 
-    service_request = models.ForeignKey(ServiceRequest, null=True, on_delete=models.SET_NULL)
-    dispatch_assignment = models.ForeignKey(EvacAssignment, null=True, on_delete=models.SET_NULL)
+    service_request = models.ForeignKey(ServiceRequest, null=True, on_delete=models.SET_NULL, related_name='come_on')
```
Comment (AlexMountain) [inline evac/models.py:40]: lol.

File: evac/serializers.py [hunk 5]
```diff
@@ -74,7 +84,7 @@ class Meta:
 
 class SimpleEvacAssignmentSerializer(serializers.ModelSerializer):
 
-    team_name = serializers.SerializerMethodField()
+    team_name = serializers.StringRelatedField(source='team')
```
Comment (AlexMountain) [inline evac/serializers.py:87]: Need to add a string method to the team object for this to return properly.

File: shelter/serializers.py [hunk 6]
```diff
@@ -7,12 +7,9 @@
 from animals.models import Animal
 
 class SimpleRoomSerializer(serializers.ModelSerializer):
-    animal_count = serializers.SerializerMethodField()
-    building_name = serializers.SerializerMethodField()
+    animal_count = serializers.IntegerField(read_only=True)
```
Comment (AlexMountain) [inline shelter/serializers.py:10]: This change isn't working, no `animal_count` is being return for rooms on building details.

**Comment by rmeyers4** [comment]
@AlexMountain : Thanks for reviewing, I think I fixed the things you pointed out without adding extra queries.
I checked SR search, Evac Assignment search, owner search, and animal search, as well as shelter list/details, building details, room details, and nothing is exploding at least.

**Pull request merged**